app/src/main/java/com/example/lutsak/rotation/BrowserActivity.java

```diff
@@ -0,0 +1,50 @@
+package com.example.lutsak.rotation;
+
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+
+
+public class BrowserActivity extends AppCompatActivity {
+
+    WebView webView;
+
+    @Override
+    public void onCreate(Bundle saveInstanceState){
+        super.onCreate(saveInstanceState);
+        setContentView(R.layout.layout_browser);
+
+        webView = findViewById(R.id.webView);
+        Uri data = getIntent().getData();
+        if (saveInstanceState == null) {
+            webView.loadUrl(data.toString());
+        }
+        webView.setWebViewClient(new WebClient());
+    }
+
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        webView.saveState(outState);
+    }
+
+    @Override
+    protected void onRestoreInstanceState(Bundle savedInstanceState) {
+        super.onRestoreInstanceState(savedInstanceState);
+        webView.restoreState(savedInstanceState);
+    }
+
+    public class WebClient extends WebViewClient {
```

Comment:
This class should be private static

Reply:
"Non-static field 'webView' cannot be referenced from a static context"
Can it be only private?

Reply:
Yes. There is no reason for it to be non-private.

Make it private static and deal with the compilation problems